arches/app/models/models.py

```diff
@@ -1280,6 +1252,118 @@ class Meta:
+    name = models.TextField()
+
+    def get_initial_resource_instance_lifecycle_state(self):
+        return self.resource_instance_lifecycle_states.get(is_initial_state=True)
+
+    def serialize(self, fields=None, exclude=None, **kwargs):
+        ret = JSONSerializer().handle_model(
+            self, fields=fields, exclude=exclude, **kwargs
+        )
+
+        ret["resource_instance_lifecycle_states"] = [
+            JSONSerializer().handle_model(resource_instance_lifecycle_state)
+            for resource_instance_lifecycle_state in self.resource_instance_lifecycle_states.all()
+        ]
+
+        return ret
+
+    class Meta:
+        db_table = "resource_instance_lifecycles"
+        managed = True
+
+
+class ResourceInstanceLifecycleState(models.Model):
+    id = models.UUIDField(primary_key=True, serialize=False)
+    name = I18n_TextField()
+    action_label = I18n_TextField()
+    is_initial_state = models.BooleanField(default=False)
+    can_delete_resource_instances = models.BooleanField(default=False)
+    can_edit_resource_instances = models.BooleanField(default=False)
+    resource_instance_lifecycle = models.ForeignKey(
+        on_delete=models.CASCADE,
+        to="models.ResourceInstanceLifecycle",
+        related_name="resource_instance_lifecycle_states",
+    )
+    previous_resource_instance_lifecycle_states = models.ManyToManyField(
+        "self",
+        through="ResourceInstanceLifecycleStateFromXRef",
+        symmetrical=False,
+        related_name="next_lifecycle_states",
+    )
+    next_resource_instance_lifecycle_states = models.ManyToManyField(
+        "self",
+        through="ResourceInstanceLifecycleStateToXRef",
+        symmetrical=False,
+        related_name="previous_lifecycle_states",
+    )
+
+    def serialize(self, fields=None, exclude=None, **kwargs):
+        ret = JSONSerializer().handle_model(
+            self, fields=fields, exclude=exclude, **kwargs
+        )
+
+        # for serialization we shouldn't need to recurse, 1 level down is enough
+        ret["next_resource_instance_lifecycle_states"] = [
+            JSONSerializer().handle_model(resource_instance_lifecycle_state)
+            for resource_instance_lifecycle_state in self.next_resource_instance_lifecycle_states.all()
+        ]
+        ret["previous_resource_instance_lifecycle_states"] = [
+            JSONSerializer().handle_model(resource_instance_lifecycle_state)
+            for resource_instance_lifecycle_state in self.previous_resource_instance_lifecycle_states.all()
+        ]
+
+        return ret
+
+    class Meta:
+        db_table = "resource_instance_lifecycle_states"
+        managed = True
+        constraints = [
+            UniqueConstraint(
+                fields=["resource_instance_lifecycle"],
+                condition=Q(is_initial_state=True),
+                name="unique_initial_state_per_lifecycle",
+            ),
+        ]
+
+
+class ResourceInstanceLifecycleStateFromXRef(models.Model):
```

Comment:
Fine to leave this as is since it's just a style comment, but for my own understanding, I tested out my suggestion (below) and it migrated (with a freshly generated set of migrations, since you can't alter a ManyToManyField) just fine, giving me a table called `public.ResourceInstanceLifecycleFromXRef` (and ...ToXRef). But you encountered a character limit on your version of Postgres?

```diff
diff --git a/arches/app/models/models.py b/arches/app/models/models.py
index f4da3bf287..e290005975 100644
--- a/arches/app/models/models.py
+++ b/arches/app/models/models.py
@@ -1324,13 +1324,13 @@ class ResourceInstanceLifecycleState(models.Model):
     )
     previous_resource_instance_lifecycle_states = models.ManyToManyField(
         "self",
-        through="ResourceInstanceLifecycleStateFromXRef",
+        db_table="ResourceInstanceLifecycleStateFromXRef",
         symmetrical=False,
         related_name="next_lifecycle_states",
     )
     next_resource_instance_lifecycle_states = models.ManyToManyField(
         "self",
-        through="ResourceInstanceLifecycleStateToXRef",
+        db_table="ResourceInstanceLifecycleStateToXRef",
         symmetrical=False,
         related_name="previous_lifecycle_states",
     )
@@ -1374,40 +1374,6 @@ class ResourceInstanceLifecycleState(models.Model):
         ]
 
 
-class ResourceInstanceLifecycleStateFromXRef(models.Model):
-    resource_instance_lifecycle_state_from = models.ForeignKey(
-        ResourceInstanceLifecycleState,
-        related_name="from_xref_next_lifecycle_states",
-        on_delete=models.CASCADE,
-    )
-    resource_instance_lifecycle_state_to = models.ForeignKey(
-        ResourceInstanceLifecycleState,
-        related_name="from_xref_previous_lifecycle_states",
-        on_delete=models.CASCADE,
-    )
-
-    class Meta:
-        db_table = "resource_instance_lifecycle_states_from_xref"
-        managed = True
-
-
-class ResourceInstanceLifecycleStateToXRef(models.Model):
-    resource_instance_lifecycle_state_from = models.ForeignKey(
-        ResourceInstanceLifecycleState,
-        related_name="to_xref_next_lifecycle_states",
-        on_delete=models.CASCADE,
-    )
-    resource_instance_lifecycle_state_to = models.ForeignKey(
-        ResourceInstanceLifecycleState,
-        related_name="to_xref_previous_lifecycle_states",
-        on_delete=models.CASCADE,
-    )
-
-    class Meta:
-        db_table = "resource_instance_lifecycle_states_to_xref"
-        managed = True
-
-
 class SearchComponent(models.Model):
     searchcomponentid = models.UUIDField(primary_key=True)
     name = models.TextField()
```